src/scatter_buildblock/ScatterSimulation.cxx

```diff
@@ -828,6 +828,55 @@ set_exam_info_sptr(const shared_ptr<const ExamInfo> arg)
     this->template_exam_info_sptr = arg->create_shared_clone();
 }
 
+
+bool 
+ScatterSimulation::
+get_downsample_scanner_bool() const
+{
+    return this->downsample_scanner_bool;
+}
+
+void 
+ScatterSimulation::
+set_downsample_scanner_bool(const bool arg)
+{
+    if (arg != this->get_downsample_scanner_bool())
+        this->_already_set_up = false;
+    this->downsample_scanner_bool = arg;
```

Comment:
could in principle do
```suggestion
    if (arg != this->get_downsample_scanner_bool())
        this->_already_set_up = false;
    this->downsample_scanner_bool = arg;
```
```suggestion
    if (arg != this->get_downsample_scanner_bool())
      {
        this->_already_set_up = false;
        this->downsample_scanner_bool = arg;
      }
```
which might be a  bit cleaner, but up to you

Reply:
I force push this for a cleaner git history